backend/src/business_logic_layer/map_manager.js

```diff
@@ -2,6 +2,47 @@ module.exports = function ({ mapRepository }) {
     return {
         getAllMaps: function (callback) {
             mapRepository.getAllMaps(callback)
+        },
+
+        createMap: function (callback) {
+            const filePath = './backend/src/data_access_layer/map.json'
+            const lastMapIdNumber = getLastMapIdNumber(filePath) // Returns the number of the last used map.
+            var mapId = "map_1"
+            /*
+            * Each Map Id will be "map_" and a Number after the "_". 
+            * The getLastMapIdNumber will return only the number and not the hole Id.
+            * I add 1 to the old id number. 
+            * If I get 0 as id number from the function, I create the firstMap Id with Id map_1 
+            * and send it to the mapRepository.
+            */
+            if (lastMapIdNumber)
+                mapId = "map_" + (lastMapIdNumber + 1)
+
+            const id = JSON.stringify(mapId, null, 2);
+            fs.writeFile(filePath, id, (error) => {
+                if (error) {
+                    callback('CouldNotWriteToFile', null)
+                    return
+                }
+                mapRepository.createMap(mapId, callback)
+            })
+
+        },
+
+        getLastMapId: function (filePath) {
```

Comment:
If the file is store in the data_access_layer and this is a function for retrieving the file then consider putting this function in the mapRepository